3-UsingAPIs/Week1/homework/ex1-johnWho.js

```diff
@@ -8,25 +8,27 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
 - If the Promise `rejects`, pass an error as the argument to reject with: "You 
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
-// TODO see above
-const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
-    }
+const getAnonName = firstName => {
+  const promising = new Promise((resolve, reject) => {
 
-    const fullName = `${firstName} Doe`;
+    setTimeout(() => {
+      if (!firstName)
+        reject(Error("You didn't pass in a first name!"));
 
-    callback(fullName);
-  }, 1000);
-};
+      const fullName = `${firstName} Doe`;
+      resolve(fullName);
```

Comment:
code style: because fullName is only used on this line, you could consider dropping it on line 18 and construct the full name directly.

```suggestion
      resolve(`${firstName} Doe`);
```